partition/roles/router/tasks/main.yaml

```diff
@@ -0,0 +1,88 @@
+---
+- name: configure mgmt vrf
+  include: mgmt_vrf.yaml
+  when: router_enable_mgmt_vrf
+
+- name: configure switch plane
+  include: switch_plane.yaml
+  when: ports is defined
+
+- name: flush handlers
+  meta: flush_handlers
+
+- name: install services
+  copy:
+    src: "{{ item }}"
+    dest: "/etc/systemd/system/{{ item }}"
+  notify: reload systemd
+  with_items:
+    - frr-validation@.service
+    - interfaces-validation@.service
+    - ifreload.service
+
+- name: copy lldpd configs
+  copy:
+    src: lldpd.d/
+    dest: /etc/lldpd.d/
+  notify: lldpd restart
+
+- name: check if lldpd has the correct portidsubtype setting
+  shell: lldpcli show configuration | grep subtype
+  register: lldpd_subtype_check
+  changed_when: false
+
+- name: trigger lldpd restart if portidsubtype setting is wrong
+  service:
+    name: lldpd
+    state: restarted
+  when: ("macaddress" not in lldpd_subtype_check.stdout)
+
+- name: check if metal-core is present to skip interfaces and frr rendering in this case
+  shell: systemctl list-units | grep metal-core
```

Comment:
I know you did just copy it over so and you do not have to change this, but using `service_facts` or something like this would look much cleaner to me.

Reply:
I see it the same way as you, using ansible modules is much cleaner. Changed the conditional check with `service_facts`